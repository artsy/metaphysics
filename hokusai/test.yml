version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
