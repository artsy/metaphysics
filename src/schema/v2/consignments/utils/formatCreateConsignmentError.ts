import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
