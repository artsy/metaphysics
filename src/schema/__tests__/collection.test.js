import { resolve } from "path"
import { readFileSync } from "fs"
import { runAuthenticatedQuery } from "test/utils"
import gql from "test/gql"
import { isEqual } from "lodash"

const gravityData = {
  id: "saved-artwork",
  name: "Saved Artwork",
  default: true,
  description: "",
  image_url: null,
  image_versions: null,
  private: false,
}

describe("Collections", () => {
  describe("Handles getting collection metadata", () => {
    it("returns collection metadata", async () => {
      const query = `
        {
          collection(id: "saved-artwork") {
            name
            private
            default
          }
        }
      `
      const rootValue = {
        collectionLoader: id =>
          id === "saved-artwork" && Promise.resolve(gravityData),
      }
      const data = await runAuthenticatedQuery(query, rootValue)
      expect(data).toMatchSnapshot()
    })

    it("returns artworks for a collection", async () => {
      const query = gql`
        {
          collection(id: "saved-artwork") {
            artworks_connection(first: 10) {
              edges {
                node {
                  id
                  title
                }
              }
            }
          }
        }
      `
      const artworksPath = resolve(
        "src",
        "test",
        "fixtures",
        "gravity",
        "artworks_array.json"
      )
      const artworks = JSON.parse(readFileSync(artworksPath, "utf8"))
      const rootValue = {
        collectionArtworksLoader: (id, params) => {
          if (
            id === "saved-artwork" &&
            isEqual(params, {
              total_count: true,
              size: 10,
              offset: 0,
              private: false,
              sort: "-position",
            })
          ) {
            return Promise.resolve({
              body: artworks,
              headers: { "x-total-count": 10 },
            })
          }

          return undefined // make undefined return explicit
        },
      }
      const data = await runAuthenticatedQuery(query, rootValue)
      expect(data).toMatchSnapshot()
    })

    it("ignores errors from gravity and returns an empty collection instead", async () => {
      const query = gql`
        {
          collection(id: "saved-artwork") {
            artworks_connection(first: 10) {
              edges {
                node {
                  id
                  title
                }
              }
            }
          }
        }
      `
      const rootValue = {
        collectionArtworksLoader: () =>
          Promise.reject(new Error("Collection Not Found")),
      }
      const {
        collection: { artworks_connection: { edges } },
      } = await runAuthenticatedQuery(query, rootValue)
      expect(edges).toEqual([])
    })
  })
})
