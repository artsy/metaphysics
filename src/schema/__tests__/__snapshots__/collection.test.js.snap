// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collections Handles getting collection metadata returns artworks for a collection 1`] = `
Object {
  "collection": Object {
    "artworks_connection": Object {
      "edges": Array [
        Object {
          "node": Object {
            "id": "nino-cais-pera-do-vento-wind-opera",
            "title": "Ã“pera do vento [Wind opera]",
          },
        },
        Object {
          "node": Object {
            "id": "ivan-navarro-tuning",
            "title": "Tuning",
          },
        },
        Object {
          "node": Object {
            "id": "jana-paleckova-untitled-2",
            "title": "Untitled",
          },
        },
        Object {
          "node": Object {
            "id": "jana-paleckova-untitled-one-eyed-skull",
            "title": "Untitled (One Eyed Skull)",
          },
        },
      ],
    },
  },
}
`;

exports[`Collections Handles getting collection metadata returns collection metadata 1`] = `
Object {
  "collection": Object {
    "default": true,
    "name": "Saved Artwork",
    "private": false,
  },
}
`;
